frame/alliance/src/lib.rs

```diff
@@ -166,30 +166,41 @@ impl<AccountId> IdentityVerifier<AccountId> for () {
 
 /// The provider of a collective action interface, for example an instance of `pallet-collective`.
 pub trait ProposalProvider<AccountId, Hash, Proposal> {
+	/// Add a new proposal.
+	/// Returns a proposal length and active proposals count if successful.
 	fn propose_proposal(
 		who: AccountId,
 		threshold: u32,
 		proposal: Box<Proposal>,
 		length_bound: u32,
 	) -> Result<(u32, u32), DispatchError>;
 
+	/// Add an aye or nay vote for the sender to the given proposal.
+	/// Returns true if the sender votes first time if successful.
 	fn vote_proposal(
 		who: AccountId,
 		proposal: Hash,
 		index: ProposalIndex,
 		approve: bool,
 	) -> Result<bool, DispatchError>;
 
+	/// Veto a proposal, close, and remove it from the system, regardless of its current state.
+	/// Returns an active proposals count which includes removed proposal.
 	fn veto_proposal(proposal_hash: Hash) -> u32;
 
+	/// Close a proposal that is either approved, disapproved or whose voting period has ended.
 	fn close_proposal(
 		proposal_hash: Hash,
 		index: ProposalIndex,
 		proposal_weight_bound: Weight,
 		length_bound: u32,
 	) -> DispatchResultWithPostInfo;
 
+	/// Return a proposal of the given hash.
 	fn proposal_of(proposal_hash: Hash) -> Option<Proposal>;
+
+	/// Return hashes of all active proposals.
```

Comment:
`Vec` is an ordered data structure, is there an expectation on how these should be ordered in the return item? E.g. by order proposed, by hash (l to h or h to l).

Reply:
There is no expectations for ordering. For now pallet does not need it to be ordered.